examples/README_mcp.md

```diff
@@ -0,0 +1,127 @@
+# DSPy Model Context Protocol (MCP) Guide
```

Comment:
Examples are not actively used. Can we have a documentation page in docs/docs/tutorials?